iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -281,10 +281,8 @@ public override async Task<Message> ReceiveAsync(CancellationToken cancellationT
                         await SubscribeCloudToDeviceMessagesAsync().ConfigureAwait(true);
                     }
 
-                    bool hasMessage = await ReceiveMessageArrivalAsync(cancellationToken).ConfigureAwait(true);
-                    Message message = ProcessMessage(hasMessage);
-
-                    return message;
+                    await WaitUntilC2dMessageArrivesAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
ConfigureAwait true or false?

Reply:
I've updated the calls to dotnetty to have ConfigureAwait(true), and rest others following ConfigureAwait(false). I've also reached out to Max to get a confirmation on the reasoning behind this.